### PR TITLE
Add more process information to mac.pslist plugin

### PR DESCRIPTION
**Pull request by cpuu**
Enhance the macOS PsList plugin by including additional process information such as offset, UID, GID, and start time in the output. Update the TreeGrid columns to display the new information. 
(I have arranged the output format and order of this plugin to be in line with the existing Volatility2.)

Before
```
$ python3 vol.py -f ../contact_me mac.pslist.PsList
Volatility 3 Framework 2.4.2
Progress:  100.00               Stacking attempts finished
PID     PPID    COMM

0       0       kernel_task
1       0       launchd
35      1       UserEventAgent
38      1       kextd
39      1       fseventsd
37      1       uninstalld
45      1       configd
46      1       powerd
52      1       logd
```

After
```
$ python3 vol.py -f ../contact_me mac.pslist.PsList
Volatility 3 Framework 2.4.2
Progress:  100.00               Stacking attempts finished
OFFSET          NAME            PID     UID     GID     Start Time                      PPID

0xff80128bb460  kernel_task     0       0       0       2019-03-23 22:06:41.934667      0
0xff801775b838  launchd         1       0       0       2019-03-23 22:06:41.934999      0
0xff801775af48  UserEventAgent  35      0       0       2019-03-23 22:06:47.314708      1
0xff801775b3c0  kextd           38      0       0       2019-03-23 22:06:47.444847      1
0xff801775c128  fseventsd       39      0       0       2019-03-23 22:06:47.446987      1
0xff801775bcb0  uninstalld      37      0       0       2019-03-23 22:06:47.444385      1
0xff801775ce90  configd         45      0       0       2019-03-23 22:06:47.457690      1
0xff801775d308  powerd          46      0       0       2019-03-23 22:06:47.459493      1
0xff801775e4e8  logd            52      0       0       2019-03-23 22:06:47.469908      1
```

### CODE REVIEW

File: volatility3/framework/plugins/mac/pslist.py [hunk 1]
```diff
@@ -105,10 +107,20 @@ def _generator(self):
...
+            offset = format_hints.Hex(task.vol.offset)
+            name = utility.array_to_string(task.p_comm)
             pid = task.p_pid
+            uid = task.p_uid
+            gid = task.p_gid
+            start_time_seconds = task.p_start.tv_sec
+            start_time_microseconds = task.p_start.tv_usec
+            start_time = datetime.datetime.fromtimestamp(
+                start_time_seconds + start_time_microseconds / 1e6
+            )
```
Comment (digitalisx) [inline volatility3/framework/plugins/mac/pslist.py:119]: Why don't you create a new associated function on the `task` object to use that information later on? These [examples](https://github.com/volatilityfoundation/volatility3/blob/2149231bb7db3c7fc48e589a16124cca373da331/volatility3/framework/symbols/windows/extensions/__init__.py#L741) can be found in `windows.pslist`.

Comment (digitalisx) [reply volatility3/framework/plugins/mac/pslist.py:119]: And these changes will need change the `_required_framework_version`.

File: volatility3/framework/plugins/mac/pslist.py [hunk 2]
```diff
@@ -2,13 +2,15 @@
...
 
+import datetime
 import logging
 from typing import Callable, Iterable, List, Dict
 
 from volatility3.framework import renderers, interfaces, exceptions
 from volatility3.framework.configuration import requirements
 from volatility3.framework.objects import utility
 from volatility3.framework.symbols import mac
+from volatility3.framework.renderers import format_hints
```
Comment (digitalisx) [inline volatility3/framework/plugins/mac/pslist.py]: It's a NIT comment. It is better to list modules in code in [alphabetical order](https://stackoverflow.com/questions/20762662/whats-the-correct-way-to-sort-python-import-x-and-from-x-import-y-statement).

**Review by digitalisx — COMMENTED** [review]
All right, I left a few comments on this change.
I don't know if the maintainer will agree with my review, but it would be good to think about it.

**Comment by ikelos** [comment]
@cpuu please could you review the open comments on this bug so we can move ahead with merging your code?

File: volatility3/framework/plugins/mac/pslist.py (hunk 2 above)
Comment (cpuu) [reply volatility3/framework/plugins/mac/pslist.py]: "Resolved: Alphabetized the imports."

**Comment by cpuu** [comment]
> @cpuu please could you review the open comments on this bug so we can move ahead with merging your code?

I apologize for the delayed response. I have just completed the changes you suggested. Some of the suggestions were applicable and have been implemented, while others fell outside the scope of this specific proposal. I will address those suggestions separately in a future Pull Request to ensure consistency across the project. Thank you for your understanding and valuable feedback.

**Review by ikelos — COMMENTED** [review]
Looks good, thanks very much for making the changes...  5:)

**Pull request merged**